### PR TITLE
Tests for 4.1.x of PKITS

### PR DESCRIPTION
**Pull request by nick-mobilecoin**
Fill in the tests for section 4.1 of PKITS.
The tests for 4.1.4-4.1.6 have been omitted as they refer to DSA which
has been deprecated, see
https://nvlpubs.nist.gov/nistpubs/FIPS/NIST.FIPS.186-5-draft.pdf#%5B%7B%22num%22%3A72%2C%22gen%22%3A0%7D%2C%7B%22name%22%3A%22XYZ%22%7D%2C72%2C721%2Cnull%5D

### CODE REVIEW

**Comment by nick-mobilecoin** [comment]
Current dependencies on/for this PR:
* main
  * **PR #68** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/68" target="_blank">Graphite</a> 
    * **PR #69** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/69" target="_blank">Graphite</a> 
      * **PR #70** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/70" target="_blank">Graphite</a> 
        * **PR #71** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/71" target="_blank">Graphite</a> 
          * **PR #72** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/72" target="_blank">Graphite</a> 
            * **PR #73** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/73" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/73?utm_source=stack-comment).

File: verifier/src/x509/chain.rs [hunk 1]
```diff
@@ -333,4 +342,45 @@ mod pkits {
 
         assert_eq!(signing_key, expected_key);
     }
+
+    #[test]
+    fn invalid_ca_signature_test_4_1_2() {
```
Comment (nick-mobilecoin) [inline verifier/src/x509/chain.rs:347]: After finishing this section of PKITS, thikning maybe the tests go to integration tests and name each test module for the PKITS pdf section i.e. `pkits-4_1.rs` or similar.

Comment (nick-mobilecoin) [reply verifier/src/x509/chain.rs:347]: Done in #75

**Review by samdealy — APPROVED** [review]

**Comment by varsha888** [comment]
Could you give me more context on PKITS and whats 4.1.1, 4.1.2 and 4.1.3 ?

**Comment by nick-mobilecoin** [comment]
> Could you give me more context on PKITS and whats 4.1.1, 4.1.2 and 4.1.3 ?

Good call out. I was too deep into what I was doing and this PR by itself doesn't have that context.

In #75 I moved this out to an integration test and put the following at the top of the file.  Let me know if this suffices or if there should be more context in the resultant test file.
```rust
//! Test for section 4.1 of the PKI test suite.
//! https://csrc.nist.gov/projects/pki-testing
//!
//! Tests 4.1.4 - 4.1.6 are not implemented because DSA has been deprecated, see
//! https://nvlpubs.nist.gov/nistpubs/FIPS/NIST.FIPS.186-5-draft.pdf#%5B%7B%22num%22%3A72%2C%22gen%22%3A0%7D%2C%7B%22name%22%3A%22XYZ%22%7D%2C72%2C721%2Cnull%5D
```

**Review by varsha888 — APPROVED** [review]

**Comment by nick-mobilecoin** [comment]
punting on implementing x509 chain parsing logic.

**Pull request closed without merging**